add sl docs (#3392)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/sl-getting-started.md

---
id: sl-getting-started
title: Quickstart
description: "Learn how to create your first semantic model and metric."
sidebar_label: Quickstart
tags: [Metrics, Semantic Layer]
---

> This quickstart guide explains the following steps and recommends a workflow that demonstrates some of MetricFlow's essential features: 

Reviewer: for this quickstart, do we have a repo we can link to, which has the complete sample data for users?

Author: Yup I linked the Jaffle shop repo

```

Interact and test your metric using the CLI before committing it to your MetricFlow repository.

## Test and query your metrics

Follow these steps to test and query your metrics using MetricFlow:

1. Make sure you have the `metricflow` and [dbt adapter](/docs/supported-data-platforms) installed in the CLI as you're installing MetricFlow as an extension of the dbt adapter. Currently, the supported adapters are Snowflake, BigQuery, Databricks, Redshift, Postgres, and DuckDB. 
    * When you install the adapter, add `[metricflow]` at the end of the command. For example, if your adapter is Snowflake, you'll run `pip install dbt-snowflake[metricflow]`
2. Run `mf version` to see your CLI version. If you do not have the CLI installed run `pip install --upgrade metricflow`
3. Save your files and run `mf validate-configs` to validate the changes before committing them
4. Run `mf query --metrics <metric_name> --dimensions <dimension_name>` to query the metrics and dimensions you want to see in the CLI.
5. Verify that the metric values are what you expect. You can view the generated SQL if you enter `--explain` in the CLI. 
6. Then commit your changes to push them to your git repo.

> <!--## Troubleshooting

Reviewer: any common troubleshooting questions and answers we can add here to help users?

---
id: sl-getting-started
title: Getting started
description: "Learn how to create your first semantic model and metric."
sidebar_label: Getting started
tags: [Metrics, Semantic Layer]
---

This getting started page recommends a workflow to help you get started creating your first metrics. Here are the following steps you'll take:

- [Create a semantic model](#create-a-semantic-model)
- [Create your metric](#create-your-metric)
- [Test and upload your metric](#test-and-upload-your-metric)

## Prerequisites

- You use dbt Core with the [command line (CLI)](/docs/core/about-the-cli) and have a dbt project set up.
    * **Note**:  Support for dbt Cloud and integrations coming soon.
- You must have an understanding of key concepts in MetricFlow (which powers the revamped dbt Semantic Layer) like [measures](/docs/build/measures), [dimensions](/docs/build/dimensions), and [entities](/docs/build/entities) before creating your first metric. Refer to [About MetricFlow](/docs/build/about-metricflow) to learn more.
- Your dbt environment must be on [dbt version 1.6](/docs/dbt-versions/core) or higher
- You have a git repository set up and your git provider has write access enabled.

> - You have a dbt project connected to a [supported data platform](/docs/supported-data-platforms) (Snowflake, BigQuery, Databricks, Redshift, Postgres, and DuckDB). 

Reviewer 2: same thing here with DW support

> - You have a development [environment](/docs/collaborate/environments/dbt-core-environments) and development credentials set up

Reviewer 2: Hmm.. this shouldn't be a requirement. We actually don't support development environments with the Semantic Layer (yet). They'll be testing commiting metrics to production

Reviewer: gotcha, thank you for clarifying! i've removed that line and rewrote the first one to say:

> -  Your dbt production environment must be on [dbt version 1.6](/docs/dbt-versions/core) or higher. Support for the development environment coming soon.

> - You have installed the [MetricFlow CLI package](https://github.com/dbt-labs/metricflow)

Reviewer 2: Note that technically they can build metrics without having the CLI, so we might want to say "In order to test your metrics, which we recommend, you'll need the CLI", or something like that

:::tip 
New to dbt or metrics? Try our [Jaffle shop example project](https://github.com/dbt-labs/jaffle-sl-template) to help you get started!
:::

## Create a semantic model

In MetricFlow, which powers the dbt Semantic Layer, there are two main objects: [semantic models](/docs/build/semantic-models) and [metrics](/docs/build/metrics-overview). You can think of semantic models as nodes in your semantic graph, connected via entities as edges. MetricFlow takes semantic models defined in YAML configuration files as inputs and creates a semantic graph that you can use to query metrics. 

> This step will guide you through setting up your semantic models, which consist of [entities](/docs/build/entities), [dimensions](/docs/build/dimensions), and [measures](/docs/build/measures).

Reviewer 2: consist should be consists

> This step will guide you through setting up your semantic models, which consist of [entities](/docs/build/entities), [dimensions](/docs/build/dimensions), and [measures](/docs/build/measures).

Reviewer: ```suggestion
This step will guide you through setting up your semantic models, which consists of [entities](/docs/build/entities), [dimensions](/docs/build/dimensions), and [measures](/docs/build/measures).
```

---
id: sl-getting-started

> title: Quickstart

Reviewer: ```suggestion
title: Getting started
```

description: "Learn how to create your first semantic model and metric."

> sidebar_label: Quickstart

Reviewer: ```suggestion
sidebar_label: Getting started
```

tags: [Metrics, Semantic Layer]
---

> This quickstart guide explains the following steps and recommends a workflow that demonstrates some of the Semantic Layer’s essential features: 

Reviewer: ```suggestion
This getting started guide explains the following steps and recommends a workflow that demonstrates some of the Semantic Layer’s essential features: 
```
